### PR TITLE
calc terminal app

### PR DESCRIPTION
**Pull request by grantsydney**

### CODE REVIEW

File: calculator.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+print("Calculator")
+result = 0
+run_again = True
+
+def calculate():
+    # global keyword allows us to modify the variable outside of the current scope
+    global result
```
Comment (manisnesan) [inline calculator.py:7]: **praise**: Good to capture the purpose. Shows why `result` is defined as global

File: calculator.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    if operation == "+":
+        result = num1 + num2
+    elif operation == "-":
+        result = num1 - num2
+    elif operation == "/":
+        if num2 == 0:
+            print("Not a number. Try again!")
```
Comment (manisnesan) [inline calculator.py]: **suggestion**: May be a better error message. Here 0 is a number. So the user may be confused. A better error message such "Division by zero is undefined. Enter a non-zero number." would be better

File: calculator.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ # this can be while True. Thought changing the value of run_again to false would
+ # cause the session to end   
+while run_again:
+    operations = ["+", "-", "/", "x"]
+    operation = input("Pick an operation: +, -, /, x")
+    if operation in operations:
+        try:
+            num1 = int(input("Enter num 1: "))
+            num2 = int(input("Enter num 2: "))
+            calculate()
```
Comment (manisnesan) [inline calculator.py]: **question**: Why `operation`, `num1` and `num2` is not passed as parameters to `calculate` method

File: calculator.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+print("Calculator")
+result = 0
+run_again = True
+
+def calculate():
```
Comment (manisnesan) [inline calculator.py]: **suggestion**:  Change the method to accept the parameters passed.
```suggestion
def calculate(operation: str, num1: int, num2: int):
```

File: calculator.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        else:
+            result = num1 / num2
+    elif operation == "x":
+        result = num1 * num2
+    print("Result:", result)
+
+ # this can be while True. Thought changing the value of run_again to false would
+ # cause the session to end   
+while run_again:
+    operations = ["+", "-", "/", "x"]
```
Comment (manisnesan) [inline calculator.py:25]: **praise**: good use of list. storing all the operations handled by the program.

File: calculator.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    operation = input("Pick an operation: +, -, /, x")
+    if operation in operations:
```
Comment (manisnesan) [inline calculator.py:27]: **praise**: Good. that's the pythonic way of doing things.

**Review by manisnesan — CHANGES_REQUESTED** [review]

File: calculator.py (hunk 3 above)
Comment (grantsydney) [reply calculator.py]: good point, thank you

**Comment by manisnesan** [comment]
Looks good to me. Proceed with the merge.

**Pull request merged**